_posts/2023-10-11-mcu-periphral-forwarding.md

```diff
@@ -60,36 +60,36 @@ int main() {
 }
 ```
 
-While this code might build successfully on our PC, executing it directly fails. The reason being, the PC application can't access the memory address 0x40011000 directly, as it's specific to the STM32F103 microcontroller's GPIOC peripheral.
+While this code might build successfully on our PC, directly executing it fails. The PC application can't directly access the memory address 0x40011000, as it's specific to the STM32F103 microcontroller's GPIOC peripheral.
 
-The challenge at hand is to enable the program to remain valid and successfully retrieve the state from a real STM32F103 chip without altering the existing codebase.
+The challenge is to enable the program to remain valid and successfully retrieve the state from a real STM32F103 chip without altering the existing codebase.
 Fortunately, the PC platform offers powerful Dynamic Binary Instrumentation (DBI) tools, empowering developers to dynamically intercept and modify memory operations during runtime.
 In the context of peripheral operations, these DBI tools become instrumental. When a memory operation targeting the peripheral address occurs, these tools can intercept the operation.
 
 ```c
 int pin_PC0_value = GPIOC->IDR & 0x1;
 ```
 
-During this interception, the program can be instructed to load a value directly from the GPIOC peripheral register of the STM32F103 chip. With the aid of a debugger, for example, we can acquire the valid value from the physical hardware and seamlessly return it to the executing operation.
+During this interception, a DBI tool can load a value directly from the GPIOC peripheral register of the STM32F103 chip. With the aid of a debugger, for example, we can acquire the valid value from the physical hardware and seamlessly return it to the executing operation.
```

Comment:
Thank you for catching that mistake and making the necessary corrections.

Reply:
It was more of change for clarity -- making the language active voice rather than passive voice! I think it will help readers with comprehension